### PR TITLE
Add the transition rules to accompany #10455

### PR DESCRIPTION
**Pull request by Myoldmopar**
Pull request overview
---------------------
I merged #10455 a little quick and didn't think about the transition changes.  This adds the missing input rules description and transition code.  @Nigusse and @mjwitte could you confirm the changes, and if so I'll remove my ?? from the input rules file :)

This doesn't really change the IDD or add a new feature, but it is related to the feature  and IDD changes, so I'm marking it as such.

### CODE REVIEW

File: src/Transition/InputRulesFiles/Rules24-1-0-to-24-2-0.md [hunk 1]
```diff
@@ -24,7 +24,19 @@ The alpha/numeric A/N numbers are not as important here.
...
+Insert new field A8 which is also F8, "Heat Recovery Outlet Node Name", defaulted to blank?
+Insert new field N3 which is F12, "Heat Recovery Reference Flow Rate", defaulted to blank for autosize?
+There are three new optional heat recovery fields at the end which transition will not add.
+
+# Object Change: HeatPump:PlantLoop:EIR:Heating
+
+Insert new field A7 which is also F7, "Heat Recovery Inlet Node Name", defaulted to blank?
+Insert new field A8 which is also F8, "Heat Recovery Outlet Node Name", defaulted to blank?
+Insert new field N3 which is F12, "Heat Recovery Reference Flow Rate", defaulted to blank for autosize?
+There are three new optional heat recovery fields at the end which transition will not add.
```
Comment (Myoldmopar) [inline src/Transition/InputRulesFiles/Rules24-1-0-to-24-2-0.md:39]:
```suggestion
# Object Change: HeatPump:PlantLoop:EIR:Cooling

Heat recovery changes as found in #10455, with the transition rules followed up in #10544

Insert new field A7 which is also F7, "Heat Recovery Inlet Node Name", defaulted to blank
Insert new field A8 which is also F8, "Heat Recovery Outlet Node Name", defaulted to blank
Insert new field N3 which is F12, "Heat Recovery Reference Flow Rate", defaulted to blank for autosize
There are three new optional heat recovery fields at the end which transition will not add.

# Object Change: HeatPump:PlantLoop:EIR:Heating

Heat recovery changes as found in #10455, with the transition rules followed up in #10544

Insert new field A7 which is also F7, "Heat Recovery Inlet Node Name", defaulted to blank
Insert new field A8 which is also F8, "Heat Recovery Outlet Node Name", defaulted to blank
Insert new field N3 which is F12, "Heat Recovery Reference Flow Rate", defaulted to blank for autosize
There are three new optional heat recovery fields at the end which transition will not add.
```

Comment (Myoldmopar) [reply src/Transition/InputRulesFiles/Rules24-1-0-to-24-2-0.md:39]: If everything looks good, we can commit this suggested change and merge this in.

**Comment by Myoldmopar** [comment]
I feel it is super appropriate that the original PR was #10455 and this is #10544.

**Comment by Myoldmopar** [comment]
@Nigusse like I said on the call, you are welcome to just checkout this branch and push commits for documentation here.  Or a different branch I don't care.  

@mjwitte ..... does this affect ExpandObjects?

**Comment by Myoldmopar** [comment]
OK last call here @Nigusse.  Do you anticipate adding anything else to this PR?  I'm not saying you need to, the docs can go in a separate PR, but I'd like to get this merged to get the transition rules in place.

**Comment by Myoldmopar** [comment]
OK, I'm going to merge in these transition rules so that the E+ code/IDD matches the transition code.  @mjwitte if you think ExpandObjects needs a change, those, along with any doc changes can all go in a separate PR.

**Comment by Myoldmopar** [comment]
OH and when that happens, or in some other branch touches it, we can remove my question marks? from the input rules description file :)

**Comment by mjwitte** [comment]
ExpandObjects doesn't support HEATPUMP:PLANTLOOP:EIR:*, so it's all good.

**Pull request merged**